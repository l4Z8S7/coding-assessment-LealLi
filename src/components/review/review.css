.review {
  width: 90%;
  min-width: 360px;
  margin: auto;
  border: 2px solid #3333ab;
  border-radius: 10px;
  padding: 20px;
  box-shadow: 9px 12px #595b61;
}

.review .title {
   font-size: 24px;
   margin: 0 0 20px 0;
   padding: 20px 0 5px 0;
   border-bottom: 4px solid #3333ab;
}

.review .answer-list {
  list-style-type: none;
  width: 100%;
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-between;
}

.review .answer-list li {
  border-bottom: 1px solid #151541;
  height: min-content;
  padding: 5px;
  cursor: pointer;
}

.review .answer-list li span {
  font-size: 16px;
}

.review .btn-submit {
  border: 1px solid #3333ab;
  border-radius: 50%;
  background-color: #5dbb49;
  font-size: 20px;
  padding: 20px;
  margin-top: 20px;
}