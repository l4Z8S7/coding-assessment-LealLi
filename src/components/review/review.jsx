import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './review.css';
import * as actions from '../../actions';

 export const Review = (props) => { 
  return (
    <div className="review">
    <p className="title">Your Answers: </p>
      <ul className="answer-list">
        {
          props.answers.map((answer, index) => <li 
            key={index}
            style={answer ? null : { backgroundColor: "#e11f36" }}
            onClick={() => props.moveToQuestion(index)}
            onKeyPress={(event) => event.key === "Enter" ? props.moveToQuestion(index) : void(0)}
            tabIndex="1"
          >
            <span><b>{ index + 1 }</b> - <i>{ answer }</i></span>
          </li>)
        }
      </ul>
      <button className="btn-submit" onClick={() => props.submitAnswer(props.answers)}>Submit</button>
    </div>
  );
}

function mapStateToProps({ answers, index }) {
  const full_answers = [...answers];
  for (let i = 0; i < index; i ++) {
    if (!full_answers[i]) {
      full_answers[i] = '';
    }
  }
  return { answers: full_answers };
}

Review.propTypes = {
  answers: PropTypes.arrayOf(PropTypes.string).isRequired,
  moveToQuestion: PropTypes.func.isRequired,
  submitAnswer: PropTypes.func.isRequired
}

export default connect(mapStateToProps, actions)(Review);