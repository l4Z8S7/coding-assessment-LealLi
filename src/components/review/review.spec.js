import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { Review } from './review';

Enzyme.configure({ adapter: new Adapter });

const props = {
  answers: ['A1 of Q1'],
  moveToQuestion: () => {},
  submitAnswer: () => {}
}

describe('Review', () => {
  it('should render component matching the snapshot', () => {
    expect(toJson(shallow(<Review {...props} />))).toMatchSnapshot();
  });
});