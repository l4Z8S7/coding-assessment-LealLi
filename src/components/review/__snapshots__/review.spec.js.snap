// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render component matching the snapshot 1`] = `
<div
  className="review"
>
  <p
    className="title"
  >
    Your Answers: 
  </p>
  <ul
    className="answer-list"
  >
    <li
      key="0"
      onClick={[Function]}
      onKeyPress={[Function]}
      style={null}
      tabIndex="1"
    >
      <span>
        <b>
          1
        </b>
         - 
        <i>
          A1 of Q1
        </i>
      </span>
    </li>
  </ul>
  <button
    className="btn-submit"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
